typedb/stream/response_collector.py

```diff
@@ -54,26 +54,23 @@ class Queue(Generic[R]):
 
         def __init__(self):
             self._response_queue: queue.Queue[Response] = queue.Queue()
-            self._error: TypeDBClientException = None
 
         def get(self, block: bool) -> R:
             response = self._response_queue.get(block=block)
             if response.is_value():
                 return response.value
-            elif response.is_done():
-                self._raise_transaction_closed_error()
+            elif response.is_done() and response.error is None:
+                raise TypeDBClientException.of(TRANSACTION_CLOSED)
+            elif response.is_done() and response.error is not None:
+                raise TypeDBClientException.of_rpc(response.error)
```

Comment:
go back to the old behaviour of retrieving the error from the `Done` message stored in the queue, which will be thrown exactly once. After that, this queue will always throw `transaction closed`.